### ISSUE TITLE
Support HTTP proxy servers for pax runner provisioning [PAXRUNNER-286]

### ISSUE THREAD

**Issue by ops4j-issues** [ops4j/org.ops4j.pax.runner#309]
**[M](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:c4c7d61b-34fa-4e34-961a-dab256a3cc6f)** created **[PAXRUNNER-286](https://ops4j1.jira.com/browse/PAXRUNNER-286)** 

In my particular case, the whole company is using a proxy in order to access the internet. To be able to use pax runner, we require some sort of proxy support. In fact, we are using a ntlm authentication, so support of authentication would be perfect, but this is more of a nice to have feature compared to general proxy support.

something like:<br /> -Dhttp.proxyHost=my.proxy.host -Dhttp.proxyPort=8080 would be perfect

---

**Affects:** 1.1.1
**Fixed in:** 1.2.0
Votes: 0, Watches: 0

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

We can support all of this http/https/ftp proxies in teh way you described (without the -D) but just a reminder you can do that also now by using environment variable JAVA_OPTS=-Dhttp.proxyHost=my.proxy.host -Dhttp.proxyPort=8080

**Comment by ops4j-issues** [comment]
**[M](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:c4c7d61b-34fa-4e34-961a-dab256a3cc6f)** commented

I tried it, but got the following error:



```java
pax-run.bat http.proxyHost=localhost http.proxyPort=5865
    ______  ________  __  __
   / __  / /  __   / / / / /
  /  ___/ /  __   / _\ \ _/
 /  /    /  / /  / / _\ \
/__/    /__/ /__/ /_/ /_/
Pax Runner (1.1.1) from OPS4J - http://www.ops4j.org
----------------------------------------------------
 -> Using config [classpath:META-INF/runner.properties]
 -> Using arguments from command line and file:/C:/wsp/pax/runner.args
 -> Scan bundles from [http.proxyHost]
 -> Scan bundles from [scan-file:http.proxyHost]
         ___
        /  /
       /  / Oops, there has been a problem!
      /  /
     /__/   org.ops4j.pax.runner.provision.ScannerException: Could not parse the provision file
    ___
   /__/     Use --log=debug to see details.
```



What am I doing wrong?

**Comment by ops4j-issues** [comment]
**[M](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:c4c7d61b-34fa-4e34-961a-dab256a3cc6f)** commented

Sorry, my fault! I didn't notice, that it is NOT implemented yet. With the JAVA_OPTS it works great. Thanks a lot for the help!

**Comment by ops4j-issues** [comment]
**[M](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:c4c7d61b-34fa-4e34-961a-dab256a3cc6f)** commented

Concerning the question if it is worth implementing. I think so, because the problem with JAVA_OPTS is that it might be a problem setting them on systems, where you have several deployment environments (some behind a proxy, some not). There you always have to reset the parameters. With a command line add-on, you are able to create a configuration for each scenario and run them in parallel. Currently, for me the JAVA_OPTS solution will do, but in general I would prefer to have it as a command line parameter, but that's just my view. Anyway, thanks again for your help!

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

You can set now via command line the http/https/ftp/sock properties (as in [http://java.sun.com/javase/6/docs/technotes/guides/net/proxies.html](http://java.sun.com/javase/6/docs/technotes/guides/net/proxies.html)).<br /> However, using this properties while embedding can have side effects as I do not backup the original props in order to restore them.



Can you test them? You will have to do something like:



```java
pax-run --http.prxyHost=myHost --http.proxyPort=9090
```

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

I added documentation in wiki: [http://wiki.ops4j.org//x/fQu4](http://wiki.ops4j.org//x/fQu4)

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

Released as part of 1.2.0
